### PR TITLE
Md/issue 1657

### PR DESCRIPTION
**Pull request by Ardiea**
# What are the relevant tickets?
https://github.com/mitodl/ol-infrastructure/issues/1657

# Description (What does it do?)
Initial creation of heroku proxy configurations items for mitopen.

### CODE REVIEW

File: pillar/heroku/mit_open.sls [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    SOCIAL_AUTH_SAML_IDP_ATTRIBUTE_NAME: urn:oid:2.16.840.1.113730.3.1.241
+    SOCIAL_AUTH_SAML_IDP_ATTRIBUTE_PERM_ID: urn:oid:1.3.6.1.4.1.5923.1.1.1.6
+    SOCIAL_AUTH_SAML_IDP_ENTITY_ID: https://idp.mit.edu/shibboleth
+    SOCIAL_AUTH_SAML_IDP_URL: https://idp.mit.edu/idp/profile/SAML2/Redirect/SSO
+    SOCIAL_AUTH_SAML_IDP_X509: __vault__::secret-operations/{{ env_data.vault_env_path }}/{{ business_unit }}/saml>data>idp_x509
+    SOCIAL_AUTH_SAML_ORG_DISPLAYNAME: MIT Office of Digital Learning
+    SOCIAL_AUTH_SAML_SECURITY_ENCRYPTED: True
+    SOCIAL_AUTH_SAML_SP_ENTITY_ID: {{ env_data.SOCIAL_AUTH_SAML_SP_ENTITY_ID }}
+    SOCIAL_AUTH_SAML_SP_PRIVATE_KEY: __vault__::secret-operations/{{ env_data.vault_env_path }}/{{ business_unit }}/saml>data>private_key
+    SOCIAL_AUTH_SAML_SP_PUBLIC_CERT: __vault__::secret-operations/{{ env_data.vault_env_path }}/{{ business_unit }}/saml>data>public_cert
```
Comment (rhysyngsun) [inline pillar/heroku/mit_open.sls:217]: These can be removed.

File: pillar/heroku/mit_open.sls [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    MAILGUN_SENDER_DOMAIN: {{ env_data.MAILGUN_SENDER_DOMAIN }}
+    MAILGUN_URL: https://api.mailgun.net/v3/{{ env_data.MAILGUN_SENDER_DOMAIN }}
+    MITPE_BASE_URL: https://professional.mit.edu/
+    EDX_LEARNING_COURSE_BUCKET_NAME: {{ env_data.EDX_LEARNING_COURSE_BUCKET_NAME }}
+    MITX_ONLINE_BASE_URL: https://mitxonline.mit.edu/
+    MITX_ONLINE_COURSES_API_URL: https://mitxonline.mit.edu/api/courses/
+    MITX_ONLINE_PROGRAMS_API_URL: https://mitxonline.mit.edu/api/programs/
+    MITX_ONLINE_LEARNING_COURSE_BUCKET_NAME: mitx-etl-mitxonline-production
+    MIT_WS_CERTIFICATE: __vault__::secret-{{ business_unit }}/global/mit-application-certificate>data>certificate
+    MIT_WS_PRIVATE_KEY: __vault__::secret-{{ business_unit }}/global/mit-application-certificate>data>private_key
```
Comment (rhysyngsun) [inline pillar/heroku/mit_open.sls:150]: These can be removed.

File: pillar/heroku/mit_open.sls [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    FEATURE_SPAM_EXEMPTIONS: True
+    FEATURE_INDEX_UPDATES: True
+    FEATURE_MOIRA: True
+    FEATURE_PODCAST_APIS: True
+    FEATURE_PODCAST_SEARCH: True
+    FEATURE_PROFILE_UI: True
+    FEATURE_SAML_AUTH: True
+    FEATURE_SEARCH_UI: True
+    FEATURE_USE_NEW_BRANDING: True
+    FEATURE_WIDGETS_UI: True
```
Comment (rhysyngsun) [inline pillar/heroku/mit_open.sls:134]: These can be removed.

File: pillar/heroku/mit_open.sls [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+  proxytype: heroku
+
+heroku:
+  app_name: {{ env_data.app_name }}
+  api_key: __vault__::secret-operations/global/heroku/odl-devops-api-key>data>value
+  config_vars:
+    AKISMET_API_KEY: __vault__::secret-{{ business_unit }}/global/akismet>data>api_key
+    AKISMET_BLOG_URL: https://discussions-rc.odl.mit.edu
+    ALGOLIA_API_KEY: __vault__::secret-operations/{{ env_data.vault_env_path }}/{{ business_unit }}/algolia>data>api_key
+    ALGOLIA_APP_ID: __vault__::secret-operations/{{ env_data.vault_env_path }}/{{ business_unit }}/algolia>data>app_id
```
Comment (rhysyngsun) [inline pillar/heroku/mit_open.sls:101]: These can be removed.

**Review by rhysyngsun — CHANGES_REQUESTED** [review]

File: pillar/heroku/mit_open.sls (hunk 3 above)
Comment (Ferdi) [reply pillar/heroku/mit_open.sls:134]: all of them ?

Comment (Ardiea) [reply pillar/heroku/mit_open.sls]: I removed all of FEATURE_* but they can easily be restored if it turns out they are needed.

**Review by blarghmatey — APPROVED** [review]

File: pillar/heroku/mit_open.sls [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+heroku:
+  app_name: {{ env_data.app_name }}
+  api_key: __vault__::secret-operations/global/heroku/odl-devops-api-key>data>value
+  config_vars:
+    AKISMET_API_KEY: __vault__::secret-{{ business_unit }}/global/akismet>data>api_key
+    AKISMET_BLOG_URL: https://discussions-rc.odl.mit.edu
+    ALLOWED_HOSTS: '["*"]'
+    AWS_ACCESS_KEY_ID:  __vault__:cache:aws/creds/mitopen>data>access_key
+    AWS_SECRET_ACCESS_KEY: __vault__:cache:aws/creds/mitopen>data>secret_key
+    AWS_STORAGE_BUCKET_NAME: 'ol-mitopen-{{ env_data.env_name }}'
```
Comment (blarghmatey) [inline pillar/heroku/mit_open.sls:103]:
```suggestion
    AWS_STORAGE_BUCKET_NAME: 'ol-mitopen-app-storage-{{ env_data.env_name }}'
```

**Pull request merged**